src/FSharpVSPowerTools.Logic/SyntaxConstructClassifier.fs

```diff
@@ -447,27 +451,24 @@ type SyntaxConstructClassifier
         | FastStage.Updating (Some { FastStageData.Snapshot = snapshot; Spans = spans }, _) ->
             let spanStartLine = max 0 (targetSnapshotSpan.Start.GetContainingLine().LineNumber - 10)
             let spanEndLine = targetSnapshotSpan.End.GetContainingLine().LineNumber + 10
-            let spans =
-                spans
-                // Locations are sorted, so we can safely filter them efficiently
-                |> Seq.skipWhile (fun span -> span.WordSpan.Line < spanStartLine)
-                |> Seq.choose (fun columnSpan -> 
-                    maybe {
-                        let! clType = getClassificationType columnSpan.Category
-                        // Create a span on the original snapshot
-                        let origSnapshot = columnSpan.Snapshot |> Option.getOrElse snapshot
-                        let! span = fromRange origSnapshot (columnSpan.WordSpan.ToRange())
-                        let span = 
-                            if targetSnapshotSpan.Snapshot <> span.Snapshot then
-                                span.TranslateTo(targetSnapshotSpan.Snapshot, SpanTrackingMode.EdgeExclusive)  
-                            else span
-                        // Translate the span to the new snapshot
-                        return clType, span 
-                    })
-                |> Seq.takeWhile (fun (_, span) -> span.Start.GetContainingLine().LineNumber <= spanEndLine)
-                |> Seq.map (fun (clType, span) -> ClassificationSpan(span, clType))
-                |> Seq.toArray
-            spans 
+            spans
+            // Locations are sorted, so we can safely filter them efficiently
+            |> Array.skipWhile (fun span -> span.WordSpan.Line < spanStartLine)
+            |> Array.choose (fun columnSpan -> 
+                maybe {
+                    let! clType = getClassificationType columnSpan.Category
+                    // Create a span on the original snapshot
+                    let origSnapshot = columnSpan.Snapshot |> Option.getOrElse snapshot
+                    let! span = fromRange origSnapshot (columnSpan.WordSpan.ToRange())
+                    let span = 
+                        if targetSnapshotSpan.Snapshot <> span.Snapshot then
+                            span.TranslateTo(targetSnapshotSpan.Snapshot, SpanTrackingMode.EdgeExclusive)  
+                        else span
+                    // Translate the span to the new snapshot
+                    return clType, span 
+                })
+            |> Array.takeWhile (fun (_, span) -> span.Start.GetContainingLine().LineNumber <= spanEndLine)
+            |> Array.map (fun (clType, span) -> ClassificationSpan (span, clType))
```

Comment:
Oh, man... This whole pipeline must be lazy (== `Seq`). Otherwise, we will `translate` all spans to new snapshot, which is very expensive. Please, revert it.